packages/react/src/utils.ts

```diff
@@ -323,14 +323,12 @@ export const disambiguateActionVariables = (
   }
 
   let newVariables: Record<string, any>;
-  const idVariable = Object.entries(action.variables).find(([key, value]) => key === "id" && value.type === "GadgetID");
 
-  if (action.acceptsModelInput || action.hasCreateOrUpdateEffect) {
+  if (action.acceptsModelInput ?? action.hasCreateOrUpdateEffect) {
```

Comment:
Hm now that I think about this a bit more, this makes less sense. `??` is nullish coalescing so `acceptsModelInput` needs to return `null` or `undefined` but it never will

Reply:
I think this will hit old apps on effect cards that have a newly generated client which has `acceptsModelInput` defined

Reply:
I think `acceptsModelInput` will be defined for newly generated api clients, but it will be undefined for old ones that are locked in time in the package.json. So, the check for `hasCreateOrUpdateEffect` is a fallback for clients that haven't been generated recently. In more recent clients though, `acceptsModelInput` is trustworthy, it correctly reports for AAC actions and the new `actionType: "custom"` stuff, as well as bulk actions, whereas the old property doesn't.